README.md

```diff
@@ -89,6 +89,34 @@ coveoPlatform.source
     });
 ```
 
+### Global Configuration
+
+It is also possible to provide global request settings for all requests made by the client; Simply provide the configuration in the `globalRequestSettings` property of the options. This can be useful for applying necessary headers or configurations for network requests. For example, you might include a proxy agent for routing requests for local development, using the `https-proxy-agent` package:
+
+#### Node.js example
+
+```js
+require('isomorphic-fetch');
+require('abortcontroller-polyfill');
+const PlatformClient = require('@coveord/platform-client').default;
+const HttpsProxyAgent = require('https-proxy-agent').default;
+
+const coveoPlatform = new PlatformClient({
+    globalRequestSettings: {
+        agent: new HttpsProxyAgent(process.env.HTTP_PROXY),
```

Comment:
`agent` doesn't seem to be a member of the type `RequestInit` 🤔

Reply:
That's correct, the `RequestInit` definition in `lib.dom.d.ts` does not include that property.

The example for Node.js relies specifically on the `RequestInit` definition from `@types/node-fetch` which would be available when polyfilling with `isomorphic-fetch` in a Node.js environment. The example is also JavaScript and is not concerned with typings.

If you would like to use a property that's available as part of the `lib.dom.d.ts` declarations I can update the docs accordingly to include a default header or something of that sort.

Reply:
```ts
import { RequestInit } from "node-fetch";
import { HttpsProxyAgent } from "https-proxy-agent";

const t: RequestInit = {
  agent: new HttpsProxyAgent("https://localhost:8888"),
};
```

Reply:
Right, although a very cool use-case, I think it would be safer to provide an example that will work for every scenario. Something very basic with the headers as you suggested would fool proof that example just fine 👍

Reply:
Example has been updated to include default headers with a reference to MDN docs for the use case.

Reply:
Example has been updated to include default headers with a reference to MDN docs for the use case :+1: